Source/WebKit/UIProcess/Model/StageModeInteractionState.h

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (C) 2025 Apple Inc. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. ``AS IS'' AND ANY
+ * EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL APPLE INC. OR
+ * CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+ * EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+ * PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+ * PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+ * OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#pragma once
+
+#if ENABLE(MODEL_PROCESS)
+
+#import <simd/simd.h>
+
+namespace WebCore {
+enum class ElementIdentifierType;
+using ElementIdentifier = ObjectIdentifier<ElementIdentifierType>;
+}
+
+namespace WebKit {
+
+struct StageModeSession {
+    WTF_MAKE_FAST_ALLOCATED;
+
+public:
+    bool isPreparingForInteraction { true };
+    simd_float4x4 transform { matrix_identity_float4x4 };
+    std::optional<WebCore::ElementIdentifier> elementID { std::nullopt };
```

Comment:
Hm... I _think_ this shouldn't be necessary, since `std::optional` will initialize to `nullopt` by default:

```suggestion
    std::optional<WebCore::ElementIdentifier> elementID;
```